test/05-supervisor-release.ts

```diff
@@ -52,4 +52,110 @@ describe('supervisor release', function() {
 			.delete(`/resin/supervisor_release(${supervisorReleaseId})`)
 			.expect(200);
 	});
+
+	describe('Devices running supervisor releases', () => {
+		let applicationId: number;
+		let device: fakeDevice.Device;
+		let small: number;
+		let middle: number;
+		let large: number;
+		before(async () => {
+			// create a new test application...
+			const { body: application } = await supertest(app, admin)
+				.post('/resin/application')
+				.send({
+					device_type: 'intel-nuc',
+					app_name: 'supervisor_release_test_app',
+				})
+				.expect(201);
+
+			applicationId = application.id;
+
+			device = await fakeDevice.provisionDevice(admin.token, applicationId);
+
+			// let's create some releases
+			const res = await supertest(app)
+				.get(`/resin/device_type?$select=id&$filter=slug eq 'intel-nuc'`)
+				.expect(200);
+
+			expect(res.body)
+				.to.have.nested.property('d[0].id')
+				.that.is.a('number');
+
+			middle = (
+				await supertest(app, admin)
+					.post(`/resin/supervisor_release`)
+					.send({
+						image_name: 'SOME_IMAGE',
+						supervisor_version: '7.0.1',
+						is_for__device_type: res.body.d[0].id,
+					})
+					.expect(201)
+			).body.id;
+
+			large = (
+				await supertest(app, admin)
+					.post(`/resin/supervisor_release`)
+					.send({
+						image_name: 'SOME_IMAGE',
+						supervisor_version: '8.0.1',
+						is_for__device_type: res.body.d[0].id,
+					})
+					.expect(201)
+			).body.id;
+
+			small = (
+				await supertest(app, admin)
+					.post(`/resin/supervisor_release`)
+					.send({
+						image_name: 'SOME_IMAGE',
+						supervisor_version: '6.0.1_logstream',
+						is_for__device_type: res.body.d[0].id,
+					})
+					.expect(201)
+			).body.id;
+		});
+
+		it('should allow setting a device to a supervisor release', async () => {
+			await supertest(app, admin)
+				.patch(`/resin/device(${device.id})`)
+				.send({
+					should_be_managed_by__supervisor_release: small,
+				})
+				.expect(200);
+		});
+
+		it('should allow updrading a devices supervisor release', async () => {
+			await supertest(app, admin)
+				.patch(`/resin/device(${device.id})`)
+				.send({
+					should_be_managed_by__supervisor_release: middle,
+				})
+				.expect(200);
+			await supertest(app, admin)
+				.patch(`/resin/device(${device.id})`)
+				.send({
+					should_be_managed_by__supervisor_release: large,
+				})
+				.expect(200);
+		});
+
+		it('should not allow downgrading a supervisor version', async () => {
+			await supertest(app, admin)
+				.patch(`/resin/device(${device.id})`)
+				.send({
+					should_be_managed_by__supervisor_release: middle,
+				})
+				.expect(400);
+		});
+
+		it('should correctly determine logstream values', async () => {
```

Comment:
This test is just doing a downgrade check when the target contains `_logstream` -- it would be good if we could add a test to make sure that `vX.Y.Z` can be upgraded to `vX.Y.Z_logstream`